### PR TITLE
Jail Filters

### LINKED ISSUE #72: Jail filters
- [ ] Settings @ Jail = Jail Filters.
- [ ] Activity Tracker 
- [ ] Faction Selector
- [ ] Time left in Jail 
- [ ] Level Filter
- [ ] Score Filter

### PR DESCRIPTION
**Pull request by Sashank999**
Closes #72 .

### CODE REVIEW

File: extension/scripts/global/globalData.js [hunk 1]
```diff
@@ -278,6 +281,16 @@ const DEFAULT_STORAGE = {
 			showAdvanced: new DefaultSetting({ type: "boolean", defaultValue: false }),
 		},
 		containers: new DefaultSetting({ type: "object", defaultValue: {} }),
+		jail: {
+			activity: new DefaultSetting({ type: "array", defaultValue: [] }),
+			faction: new DefaultSetting({ type: "string", defaultValue: "" }),
+			timeStart: new DefaultSetting({ type: "string", defaultValue: "0" }),
```
Comment (DeKleineKobini) [inline extension/scripts/global/globalData.js:287]: Use the proper type for those, they are `number`s, with `string` they'll get reset to default every time.

File: extension/scripts/global/functions/slider.css [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	z-index: 1;
+	border-radius: 50px;
+}
+
+.tt-dual-range .handle {
+	width: 26px;
+	height: 26px;
+	background: #fff;
+	position: absolute;
+	box-shadow: 0px 3px 6px 0px rgba(0, 0, 0, 0.1);
```
Comment (DeKleineKobini) [inline extension/scripts/global/functions/slider.css:25]: `0`

**Review by DeKleineKobini — CHANGES_REQUESTED** [review]
I'll leave the slider feedback to the other PR.

Score filter isn't saved at all, should be fixed.

File: extension/scripts/global/functions/slider.css (hunk 2 above)
Comment (Sashank999) [reply extension/scripts/global/functions/slider.css]: Changed.

File: extension/scripts/global/globalData.js (hunk 1 above)
Comment (Sashank999) [reply extension/scripts/global/globalData.js]: Changed.

**Comment by Sashank999** [comment]
> I'll leave the slider feedback to the other PR.
> 
> Score filter isn't saved at all, should be fixed.

It isn't meant to be saved. The score changes everytime when the player in the top of Jail List changes. And hence I didn't try to save them.
Should we really save them ?
Just like factions, we could just leave it by default from 0 to scoreMax.

File: extension/scripts/features/jailfilter/ttJailFilter.js [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+				newSlider(),
+				document.newElement({
+					type: "div",
+					class: "level-filter-info-wrap",
+					children: [document.newElement({ type: "span", class: "level-filter-info" })],
+				}),
+			],
+		});
+		const scoreMax =
+			Math.max(100 * (parseInt(document.find(".users-list > *:first-child .info-wrap .time").lastChild.textContent.trim().split(" ")[0].replace(/[hs]/g, "")) + 3), 5000);
```
Comment (DeKleineKobini) [inline extension/scripts/features/jailfilter/ttJailFilter.js:82]: Score should be in a function, should it ever change we'll only have to edit it in 1 place.

File: extension/scripts/features/jailfilter/ttJailFilter.js [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+					continue;
+				}
+				// Level
+				const level = parseInt(li.find(".info-wrap .level").innerText.replace(/\D+/g, ""));
+				if ((levelStart && level < levelStart) || (levelEnd !== 100 && level > levelEnd)) {
+					showRow(li, false);
+					continue;
+				}
+				// Score
+				const score = level * (timeLeftHrs + 3);
```
Comment (DeKleineKobini) [inline extension/scripts/features/jailfilter/ttJailFilter.js:204]: Use score function

**Review by DeKleineKobini — CHANGES_REQUESTED** [review]

**Review by DeKleineKobini — APPROVED** [review]

**Review by DeKleineKobini — APPROVED** [review]

**Pull request merged**